File: data/compilers_comparation/src/examples/second_example/src/func14.c
#include "second_example.h" 
array_t* func14(array_t_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   array_t* array211;
   if (pCounter > 0) {
      array211 = vars->data[--pCounter];
      array211->refC++;
      DEBUG_COPY(array211->id);
   } else {
      array211 = (array_t*)malloc(sizeof(array_t));
      array211->size = 563;
      array211->refC = 1;
      array211->id = 211;
      array211->data = (unsigned int*)malloc(array211->size*sizeof(unsigned int));
      memset(array211->data, 0, array211->size*sizeof(unsigned int));
      DEBUG_NEW(array211->id);
   }
   unsigned int loop61 = 0;
   unsigned int loopLimit61 = (rand()%loopsFactor)/3 + 1;
   for(; loop61 < loopLimit61; loop61++) {
      if(PATH0 & 1) {
         array_t* array212;
         if (pCounter > 0) {
            array212 = vars->data[--pCounter];
            array212->refC++;
            DEBUG_COPY(array212->id);
         } else {
            array212 = (array_t*)malloc(sizeof(array_t));
            array212->size = 49;
            array212->refC = 1;
            array212->id = 212;
            array212->data = (unsigned int*)malloc(array212->size*sizeof(unsigned int));
            memset(array212->data, 0, array212->size*sizeof(unsigned int));
            DEBUG_NEW(array212->id);
         }
         unsigned int loop62 = 0;
         unsigned int loopLimit62 = (rand()%loopsFactor)/4 + 1;
         for(; loop62 < loopLimit62; loop62++) {
            for (int i = 0; i < array211->size; i++) {
               array211->data[i]--;
            }
            for (int i = 0; i < array212->size; i++) {
               if (array212->data[i] == 5) { 
                  return array212;
               }
            }
         }
         array_t_param params0;
         params0.size = 2;
         params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
         params0.data[0] = array211;
         params0.data[1] = array212;
         array_t* array213 = func28(&params0, rng(), loopsFactor);
         DEBUG_RETURN(array213->id);
         free(params0.data);
         array213->refC--;
         if(array213->refC == 0) {
            free(array213->data);
            free(array213);
            DEBUG_FREE(array213->id);
         }
         array212->refC--;
         if(array212->refC == 0) {
            free(array212->data);
            free(array212);
            DEBUG_FREE(array212->id);
         }
      }
      else {
         array_t_param params0;
         params0.size = 1;
         params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
         params0.data[0] = array211;
         array_t* array214 = func22(&params0, rng(), loopsFactor);
         DEBUG_RETURN(array214->id);
         free(params0.data);
         array214->refC--;
         if(array214->refC == 0) {
            free(array214->data);
            free(array214);
            DEBUG_FREE(array214->id);
         }
      }
      array_t* array215;
      if (pCounter > 0) {
         array215 = vars->data[--pCounter];
         array215->refC++;
         DEBUG_COPY(array215->id);
      } else {
         array215 = (array_t*)malloc(sizeof(array_t));
         array215->size = 614;
         array215->refC = 1;
         array215->id = 215;
         array215->data = (unsigned int*)malloc(array215->size*sizeof(unsigned int));
         memset(array215->data, 0, array215->size*sizeof(unsigned int));
         DEBUG_NEW(array215->id);
      }
      array_t_param params0;
      params0.size = 2;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      params0.data[0] = array211;
      params0.data[1] = array215;
      array_t* array216 = func23(&params0, loopsFactor);
      DEBUG_RETURN(array216->id);
      free(params0.data);
      if(PATH0 & 2) {
         array_t* array217;
         if (pCounter > 0) {
            array217 = vars->data[--pCounter];
            array217->refC++;
            DEBUG_COPY(array217->id);
         } else {
            array217 = (array_t*)malloc(sizeof(array_t));
            array217->size = 993;
            array217->refC = 1;
            array217->id = 217;
            array217->data = (unsigned int*)malloc(array217->size*sizeof(unsigned int));
            memset(array217->data, 0, array217->size*sizeof(unsigned int));
            DEBUG_NEW(array217->id);
         }
         unsigned int loop63 = 0;
         unsigned int loopLimit63 = (rand()%loopsFactor)/4 + 1;
         for(; loop63 < loopLimit63; loop63++) {
            for (int i = 0; i < array216->size; i++) {
               array216->data[i]--;
            }
            for (int i = 0; i < array215->size; i++) {
               if (array215->data[i] == 82) { 
                  return array215;
               }
            }
         }
         array_t_param params1;
         params1.size = 4;
         params1.data = (array_t**)malloc(params1.size*sizeof(array_t*));
         params1.data[0] = array211;
         params1.data[1] = array215;
         params1.data[2] = array216;
         params1.data[3] = array217;
         array_t* array218 = func36(&params1, rng(), loopsFactor);
         DEBUG_RETURN(array218->id);
         free(params1.data);
         array218->refC--;
         if(array218->refC == 0) {
            free(array218->data);
            free(array218);
            DEBUG_FREE(array218->id);
         }
         array217->refC--;
         if(array217->refC == 0) {
            free(array217->data);
            free(array217);
            DEBUG_FREE(array217->id);
         }
      }
      else {
         array_t_param params1;
         params1.size = 3;
         params1.data = (array_t**)malloc(params1.size*sizeof(array_t*));
         params1.data[0] = array211;
         params1.data[1] = array215;
         params1.data[2] = array216;
         array_t* array219 = func32(&params1, rng(), loopsFactor);
         DEBUG_RETURN(array219->id);
         free(params1.data);
         array219->refC--;
         if(array219->refC == 0) {
            free(array219->data);
            free(array219);
            DEBUG_FREE(array219->id);
         }
      }
      for (int i = 0; i < array211->size; i++) {
         if (array211->data[i] == 74) { 
            return array211;
         }
      }
      array216->refC--;
      if(array216->refC == 0) {
         free(array216->data);
         free(array216);
         DEBUG_FREE(array216->id);
      }
      array215->refC--;
      if(array215->refC == 0) {
         free(array215->data);
         free(array215);
         DEBUG_FREE(array215->id);
      }
   }
   for (int i = 0; i < array211->size; i++) {
      array211->data[i]--;
   }
   for (int i = 0; i < array211->size; i++) {
      if (array211->data[i] == 86) { 
         return array211;
      }
   }
   return array211;
}

File: data/multilang_comparison/program_go/program_go/src/func14.go
package main
import "math/rand"
var _ = rand.Intn
func Func14(vars *ArrayParam, PATH0 uint64 , loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   var array179 *Array
   if pCounter > 0 {
       array179 = vars.Data[pCounter-1]
       pCounter--
       array179.RefC++
   } else {
       array179 = &Array{
           Size: 237,
           RefC: 1,
           Id: 179,
           Data: make([]uint32, 237),
       }
   }
   var loop52 int = 0;
   var loopLimit52 int = (50)/3 + 1;
   for ; loop52 < loopLimit52; loop52++ {
      if (PATH0 & 1) != 0 {
         var array180 *Array
         if pCounter > 0 {
             array180 = vars.Data[pCounter-1]
             pCounter--
             array180.RefC++
         } else {
             array180 = &Array{
                 Size: 205,
                 RefC: 1,
                 Id: 180,
                 Data: make([]uint32, 205),
             }
         }
         var loop53 int = 0;
         var loopLimit53 int = (50)/4 + 1;
         for ; loop53 < loopLimit53; loop53++ {
            for i := 0; i < array180.Size; i++ {
                array180.Data[i]--
            }
            for i := 0; i < array180.Size; i++ {
                if array180.Data[i] == 18 {
                    return array180
                }
            }
         }
         var params0 ArrayParam
         params0.Size = 2
         params0.Data = make([]*Array, params0.Size)
         params0.Data[0] = array179
         params0.Data[1] = array180
         var array181 *Array = Func28(&params0, getPath(), loopsFactor);
         params0.Data = nil
         params0.Size = 0
         array181.RefC--
         if array181.RefC == 0 {
             array181.Data = nil
             array181 = nil
         }
         array180.RefC--
         if array180.RefC == 0 {
             array180.Data = nil
             array180 = nil
         }
      
      } else {
         var params0 ArrayParam
         params0.Size = 1
         params0.Data = make([]*Array, params0.Size)
         params0.Data[0] = array179
         var array182 *Array = Func22(&params0, getPath(), loopsFactor);
         params0.Data = nil
         params0.Size = 0
         array182.RefC--
         if array182.RefC == 0 {
             array182.Data = nil
             array182 = nil
         }
      }
      var array183 *Array
      if pCounter > 0 {
          array183 = vars.Data[pCounter-1]
          pCounter--
          array183.RefC++
      } else {
          array183 = &Array{
              Size: 873,
              RefC: 1,
              Id: 183,
              Data: make([]uint32, 873),
          }
      }
      var params0 ArrayParam
      params0.Size = 2
      params0.Data = make([]*Array, params0.Size)
      params0.Data[0] = array179
      params0.Data[1] = array183
      var array184 *Array = Func23(&params0, loopsFactor);
      params0.Data = nil
      params0.Size = 0
      if (PATH0 & 2) != 0 {
         var array185 *Array
         if pCounter > 0 {
             array185 = vars.Data[pCounter-1]
             pCounter--
             array185.RefC++
         } else {
             array185 = &Array{
                 Size: 483,
                 RefC: 1,
                 Id: 185,
                 Data: make([]uint32, 483),
             }
         }
         var loop54 int = 0;
         var loopLimit54 int = (50)/4 + 1;
         for ; loop54 < loopLimit54; loop54++ {
            for i := 0; i < array185.Size; i++ {
                array185.Data[i]--
            }
            for i := 0; i < array184.Size; i++ {
                if array184.Data[i] == 57 {
                    return array184
                }
            }
         }
         var params1 ArrayParam
         params1.Size = 4
         params1.Data = make([]*Array, params1.Size)
         params1.Data[0] = array179
         params1.Data[1] = array183
         params1.Data[2] = array184
         params1.Data[3] = array185
         var array186 *Array = Func36(&params1, getPath(), loopsFactor);
         params1.Data = nil
         params1.Size = 0
         array186.RefC--
         if array186.RefC == 0 {
             array186.Data = nil
             array186 = nil
         }
         array185.RefC--
         if array185.RefC == 0 {
             array185.Data = nil
             array185 = nil
         }
      
      } else {
         var params1 ArrayParam
         params1.Size = 3
         params1.Data = make([]*Array, params1.Size)
         params1.Data[0] = array179
         params1.Data[1] = array183
         params1.Data[2] = array184
         var array187 *Array = Func32(&params1, getPath(), loopsFactor);
         params1.Data = nil
         params1.Size = 0
         array187.RefC--
         if array187.RefC == 0 {
             array187.Data = nil
             array187 = nil
         }
      }
      for i := 0; i < array183.Size; i++ {
          if array183.Data[i] == 18 {
              return array183
          }
      }
      array184.RefC--
      if array184.RefC == 0 {
          array184.Data = nil
          array184 = nil
      }
      array183.RefC--
      if array183.RefC == 0 {
          array183.Data = nil
          array183 = nil
      }
   }
   for i := 0; i < array179.Size; i++ {
       array179.Data[i]--
   }
   for i := 0; i < array179.Size; i++ {
       if array179.Data[i] == 88 {
           return array179
       }
   }
   return array179;
}

